community/logging/src/main/java/org/neo4j/logging/RotatingFileOutputStreamSupplier.java

```diff
@@ -190,40 +231,58 @@ private void rotate()
             return;
         }
 
-        Runnable runnable = () -> {
-            OutputStream newStream;
-            try
+        Runnable runnable = () ->
+        {
+            synchronized ( outRef )
```

Comment:
We never read from the log files though, we only write to them. So it would be write locks everywhere.